docs/framework/architecture/ui-library.md

```diff
@@ -439,6 +441,170 @@ dropdownView.bind( 'isEnabled' ).toMany( buttons, 'isEnabled',
+
+##### Header
+
+A header may consist of any combination of three elements:
+
+* the icon,
+* the title,
+* the "Close" button.
+
+The "Close" button is added automatically as long as you provide an icon or a title. The code below shows how to create display a dialog with a header that comes with an icon and a title, but without the "Close" button.
+
+```js
+import { icons } from 'ckeditor5/src/core.js';
+
+// ...
+
+dialog.show( {
+	icon: icons.pencil,
+	title: 'My first dialog',
+	// Do not display the "Close" button.
+	hasCloseButton: false,
+
+	// The rest of the dialog definition...
+} );
+```
+
+##### Content
+
+This part can be a single [view](#views) or a collection of views. They will be displayed directly inside a body of a dialog.
+
+##### Action buttons
+
+The last piece of the dialog is the actions area, where the buttons are displayed. Their behaviour can be fully customized using the `onCreate()` and `onExecute()` callbacks. Below you can find an example of the configuration for the four custom buttons:
+
+* The "OK" button that closes the dialog and has a custom CSS class set,
+* The "Set custom title" button that changes the dialog's title,
+* The "Drag detector" button that changes its state once the dialog was dragged by the user.
+* The "Cancel" button that closes the dialog.
+
+```js
+dialog.show( {
+	actionButtons: [
+		{
+			label: 'OK',
+			class: 'ck-button-action',
+			withText: true,
+			onExecute: () => dialog.hide()
+		},
+		{
+			label: 'Set custom title',
+			withText: true,
+			onExecute: () => {
+				dialog.view!.headerView!.label = 'New title';
+			}
+		},
+		{
+			label: 'Drag detector'
+			withText: true,
+			onCreate: buttonView => {
+				buttonView.bind( 'isEnabled' ).to( dialog.view!, 'wasMoved', wasMoved => !wasMoved );
+			}
+		},
+		{
+			label: 'Cancel',
+			withText: true,
+			onExecute: () => dialog.hide()
+		}
+	]
+} );
+```
+
+#### Accessibility
+
+Dialogs provide full keyboard accessibility. While a dialog is open, press the <kbd>Ctrl</kbd>+<kbd>F6</kbd> combination to move the focus between the editor and the dialog. It can also be closed at any time by pressing the <kbd>Esc</kbd> key. To navigate through the dialog, use the <kbd>Tab</kbd> and <kbd>Shift</kbd>+<kbd>Tab</kbd> keystrokes. The contents of the dialog are also available for screen readers.
+
+#### API
+
+The dialog's lifecycle (creating and destroying) is managed by the {@link module:ui/dialog/dialog Dialog plugin}. It provides two public methods: {@link module:ui/dialog/dialog~Dialog#show `show()`} and {@link module:ui/dialog/dialog~Dialog#hide `hide()`}. Both of them fire the respective events ({@link module:ui/dialog/dialog~DialogShowEvent `show`} and {@link module:ui/dialog/dialog~DialogHideEvent `hide`}), so it's possible to hook after or before them.
+
```

Comment:
It could be just in API docs but maybe it would be worth mentioning that the dialog is opened on `normal` priority and that `onShow` and `onHide` callbacks are called on `low` priority.

Also, now as I think of it, do we even need `onShow` and `onHide` if we have `show` and `hide` events? I guess they are simpler to use but I am getting redundancy smell. And then somebody reads guide and API docs and is confused when they should use this and when that. WDYT? cc @oleq

Reply:
At some point, we decided to have both:
* callbacks for convenience: 90% of users will be satisfied by them and they won't need to dive into our events system, they are straightforward and correspond to the API in other dialog libraries.
* events for power users who want more